theano/sandbox/gpuarray/extra_ops.py

```diff
@@ -0,0 +1,489 @@
+from __future__ import absolute_import, print_function, division
+import theano
+import numpy
+import os
+from theano import Op, Apply, config
+from theano.tensor.extra_ops import CumsumOp
+
+try:
+    import pygpu
+    from pygpu import gpuarray
+except ImportError:
+    pass
+
+from .basic_ops import (as_gpuarray_variable, GpuKernelBase, Kernel,
+                        infer_context_name, GpuFromHost, HideC)
+from .opt import register_opt as register_gpu_opt, op_lifter
+from .type import GpuArrayType
+
+
+class GpuCumsum(GpuKernelBase, HideC, CumsumOp):
```

Comment:
Since you are not actually using anything from CumsumOp, you can probably remove it and HideC from the superclasses.